code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -46,4 +44,6 @@ const StageProgress = React.createClass({
     );
   }
 });
-module.exports = StageProgress;
+export default connect((state, ownProps) => ({
+  levels: ownProps.levels || state.stages[0].levels
```

Comment:
Interesting.  This feels a little like giving the `StageProgress` component more knowledge of the context in which it's used than it should have.  I might go with leaving this component unconnected and always expecting `levels` to get passed in.  If we're rending StageProgress directly in some cases, you could have a thin wrapper component be connected to Redux in that case - that way the thing that calls `state.stages[0].levels` is _only_ used in that header case.

Alternatively, a comment explaining this might be fine.